Modules/_datetimemodule.c

```diff
@@ -6386,18 +6386,35 @@ PyInit__datetime(void)
     if (m == NULL)
         return NULL;
 
-    if (PyType_Ready(&PyDateTime_DateType) < 0)
+    if (PyModule_AddType(m, &PyDateTime_DateType) < 0) {
+        Py_DECREF(m);
         return NULL;
-    if (PyType_Ready(&PyDateTime_DateTimeType) < 0)
+    }
+
+    if (PyModule_AddType(m, &PyDateTime_DateTimeType) < 0) {
+        Py_DECREF(m);
         return NULL;
-    if (PyType_Ready(&PyDateTime_DeltaType) < 0)
+    }
+
+    if (PyModule_AddType(m, &PyDateTime_TimeType) < 0) {
+        Py_DECREF(m);
         return NULL;
-    if (PyType_Ready(&PyDateTime_TimeType) < 0)
+    }
+
+    if (PyModule_AddType(m, &PyDateTime_DeltaType) < 0) {
+        Py_DECREF(m);
         return NULL;
-    if (PyType_Ready(&PyDateTime_TZInfoType) < 0)
+    }
+
+    if (PyModule_AddType(m, &PyDateTime_TZInfoType) < 0) {
+        Py_DECREF(m);
```

Comment:
I  second @pganssle re: `Py_DECREF` on error.

`PyModule_AddType` co go three ways:
* no change to refcount, error
* incref+decref, rerror
* incref, OK

So, my take on usage of `PyModule_AddType` is that it should be:

`if (PyModule_AddType(...) < 0) { return NULL; }`

That is, you'd only decref on error if you've incref'd first. This module never incref's, so it should also never decref these members.

Reply:
An example that doesn't decref module on error:
https://github.com/python/cpython/blob/188078c39dec24aa5b3f2073bdc9a68ebaae42de/Modules/_blake2/blake2module.c#L60-L88

An example that does:
https://github.com/python/cpython/blob/6ff79f65820031b219622faea8425edaec9a43f3/Modules/readline.c#L1343-L1373

Reply:
@pganssle @dimaqq 
Thank you for very kind reviewing.
Okay, I decide to change the reference count as same as the module did.
That looks safer for this time.

Reply:
@dimaqq: "An example that doesn't decref module on error: (...)"

Please don't look at existing code as an example how to correctly write code. Most PyInit functions misuse the Python C API in the error handling paths: most are leaking a strong reference to the module on error...